### PR TITLE
WSTEAM1-1158: Adds Inline Script for Opera Mini ATI Beacon Request

### PR DESCRIPTION
**Pull request by shayneahchoon**
Resolves JIRA [WSTEAM1-1158](https://jira.dev.bbc.co.uk/browse/WSTEAM1-1158)

Overall changes
======
- Reverts this [revert](https://github.com/bbc/simorgh/pull/11749), which will effectively implement the scripts that send beacon data regarding opera mini. -- Reverts bbc/simorgh#11749
- Merges the revert with this [PR](https://github.com/bbc/simorgh/pull/11744), which upgrades our Ramda dependencies. 

Code Changes
======
**Ramda and Ramda types upgraded to 0.30 from a patched version of 0.28**

### CODE REVIEW

**Review by amoore108 — APPROVED** [review]
We can give this a go and closely monitor the analytics to see if it affects Opera Mini.

File: src/app/components/ATIAnalytics/canonical/sendBeaconOperaMini.test.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    expect(XMLHttpRequestMock.open).toHaveBeenCalledWith(
+      'GET',
+      'https://foobar.com',
+      true,
+    );
+  });
+
+  it('should NOT send more than 1 beacon with XHR, when browser is Opera Mini', () => {
+    const check = {
+      hasOperaMinScriptRun: false,
```
Comment (karinathomasbbc) [inline src/app/components/ATIAnalytics/canonical/sendBeaconOperaMini.test.ts]: typo?
```suggestion
      hasOperaMiniScriptRun: false,
```

File: src/app/components/ATIAnalytics/canonical/sendBeaconOperaMini.test.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
+/* eslint-disable no-eval */
+import sendBeaconOperaMiniScript from './sendBeaconOperaMiniScript';
+
+interface WindowOperaMini extends Window {
+  hasOperaMinScriptRun?: boolean;
```
Comment (karinathomasbbc) [inline src/app/components/ATIAnalytics/canonical/sendBeaconOperaMini.test.ts]:
```suggestion
  hasOperaMiniScriptRun?: boolean;
```

**Review by karinathomasbbc — APPROVED** [review]
Thanks for the updates! 🤞 this works & we never need to worry about Opera Mini again 😅

**Pull request merged**